### PR TITLE
ROX-26408: Encrypt Microsoft Sentinel Notifier Secret

### PR DESCRIPTION
**Pull request by SimonBaeumer**
### Description

ROX-26408: Encrypt Microsoft Sentinel Notifier Secret

change me!

### User-facing documentation

- [ ] CHANGELOG is updated **OR** update is not needed
- [ ] [documentation PR](https://spaces.redhat.com/display/StackRox/Submitting+a+User+Documentation+Pull+Request) is created and is linked above **OR** is not needed

#### Automated testing

 - Added unit tests

### CODE REVIEW

File: central/notifiers/utils/encryption.go [hunk 1]
```diff
@@ -158,6 +158,11 @@ func getCredentials(notifier *storage.Notifier) (string, error) {
 		return notifier.GetPagerduty().GetApiKey(), nil
 	case pkgNotifiers.GenericType:
 		return notifier.GetGeneric().GetPassword(), nil
+	case pkgNotifiers.MicrosoftSentinelType:
+		if notifier.GetMicrosoftSentinel().GetSecret() != "" {
+			return notifier.GetMicrosoftSentinel().GetSecret(), nil
+		}
+		return notifier.GetMicrosoftSentinel().GetClientCertAuthConfig().GetPrivateKey(), nil
```
Comment (charmik-redhat) [inline central/notifiers/utils/encryption.go:165]: Is `PrivateKey` the only thing that we want to encrypt? Or should `ClientCert` be encrypted too? In other words, can someone do any damage if they got hands on `ClientCert`?

**Review by charmik-redhat — COMMENTED** [review]
Might have to decrypt the credentials too when the notifier is used to send alert and logs.

File: central/notifiers/utils/encryption.go (hunk 1 above)
Comment (SimonBaeumer) [reply central/notifiers/utils/encryption.go:165]: No, clientCert is the public and will be presented to all servers the notifier tries to connect to.

File: dev-tools/create-notifier-secret.sh
Comment (charmik-redhat) [inline dev-tools/create-notifier-secret.sh:1]: Do you want to keep this for future testing?

**Review by charmik-redhat — APPROVED** [review]

File: dev-tools/create-notifier-secret.sh
Comment (SimonBaeumer) [reply dev-tools/create-notifier-secret.sh:1]: Yes, I'll include it in a follow-up PR where I'll add documentation.

**Review by charmik-redhat — APPROVED** [review]
Looks good.

**Pull request merged**